pypesto/objective/history.py

```diff
@@ -20,53 +25,88 @@ class ObjectiveHistory:
     Attributes
     ----------
 
-    n_fval, n_grad, n_hess, n_res, n_sres: int
+    n_fval, n_grad, n_hess, n_res, n_sres:
         Counters of function values, gradients and hessians,
         residuals and residual sensitivities.
 
-    trace: list
+    trace:
         List containing history of function values and parameters if
         options.tr_record is True.
 
-    start_time: float
+    start_time:
         Reference start time.
 
-    fval0, fval_min: float
+    fval0, fval_min:
         Initial and best function value found.
 
-    x0, x_min: np.ndarray
+    x0, x_min:
         Initial and best parameters found.
 
-    index: str
+    x_names:
+        parameter names
+
+    index:
         Id identifying the history object when called in a multistart
         setting.
+
+    fval2chi2_offset:
+        conversion constant to convert chi2 values to fvals
+
+    obj:
+        _call_mode_fun of the parent objective (if available)
     """
 
-    def __init__(self, options=None):
+    def __init__(self,
+                 options: Optional[ObjectiveOptions] = None,
+                 x_names: Optional[List[str]] = None,
+                 obj: Optional[Callable] = None) -> None:
 
         if options is None:
             options = ObjectiveOptions()
-        self.options = options
+        self.options: Union[ObjectiveOptions, None] = options
+        self.x_names: Union[List[str], None] = x_names
 
-        self.n_fval = None
-        self.n_grad = None
-        self.n_hess = None
-        self.n_res = None
-        self.n_sres = None
+        self.n_fval: Union[int, None] = None
```

Comment:
could just use int for these, since an empty None value should be clear anyway ... but no strong opinion.

Reply:
pycharm syntax highlighting produces a warning, so I would rather keep it like this